internal/controller/management_controller.go

```diff
@@ -211,6 +218,47 @@ func (r *ManagementReconciler) Update(ctx context.Context, management *hmc.Manag
 	return ctrl.Result{}, nil
 }
 
+func (r *ManagementReconciler) startDependentControllers(ctx context.Context, management *hmc.Management) error {
+	l := ctrl.LoggerFrom(ctx)
+
+	if !r.SveltosDependentControllersStarted {
+		if management.Status.Components[hmc.ProviderSveltosName].Success {
```

Comment:
the nested blocks could be simplified
```
if started {
  return nil
}

if !mgmt.status.success {
   log
   return nil // btw, should we trigger requeue here?
}

// rest of the main logic
```


thoughts: the flag could be interchangeable with `sync.Once[Func]` or just an `atomic` but it probably might decrease readability with the former, atomic is just in case since I can't image the event in which the ctrl could run something in parallel (unless it is set explicitly by developers)

Reply:
I used `sync.OnceFunc` with panics in the latest commit. With using it, I got rid of the `SveltosDependentControllersStarted` flag. Do you agree with its use?

Reply:
i haven't noticed it yet, now i see. to me it looks okay! i'd only suggest to skip that huge `if` block as I suggested in the initial comment:

```
	if !management.Status.Components[kcm.ProviderSveltosName].Success {
            l.Info("Waiting for provider to be ready to setup contollers dependent on it", "provider", kcm.ProviderSveltosName))
            return
	}
	// syncOnce
```